### ISSUE TITLE
useTranslation is the cause of the rerender component 

### ISSUE THREAD

**Issue by zcmgyu** [i18next/react-i18next#1291]
## 🐛 Bug Report

I got rerendering issue while use const { t } = useTranslation() of react-i18next. 
In my research, the inside of useTranslation hook used useState to store a getT function. Wrapping getT function inside useCallback hook could not prevent getting this message from WDYR

> different functions with the same name

```javascript
function getT() {
    return i18n.getFixedT(null, i18nOptions.nsMode === 'fallback' ? namespaces : namespaces[0]);
}
```
https://github.com/i18next/react-i18next/blob/master/src/useTranslation.js#L42-L45

```javascript
setT(getT);
```
https://github.com/i18next/react-i18next/blob/master/src/useTranslation.js#L82

Could you please give me the idea to prevent a re-rendering issue every time use `t` of useTranslation? Thanks 🙇🏻 

## To Reproduce

Repo: https://github.com/zcmgyu/react-i18next-rerender-issue

<img width="662" alt="Screen Shot 2021-04-06 at 12 33 17" src="https://user-images.githubusercontent.com/13638569/113655197-55b39e00-96d4-11eb-8e47-dc6426635691.png">

<img width="896" alt="Screen Shot 2021-04-06 at 12 33 33" src="https://user-images.githubusercontent.com/13638569/113655208-59472500-96d4-11eb-9a1f-ce7e1e40eb93.png">

## Expected behavior

Could `useTranslation` feature without re-render because of the same name of `fixedT`

## Your Environment

- *runtime version*: Node v15.8.0, browser Chrome
- *i18next version*: ^20.1.0
- *react-i18next*: ^11.8.12 
- *os*: MacOS Big Sur

**Comment by zcmgyu** [comment]
Same issue with next-i18next

https://github.com/isaachinman/next-i18next/issues/654

**Comment by jamuhl** [comment]
check the code of useTranslation -> it sets that function initially and on languageChange (and optionally by userland code binded events) -> so that function only changes if something is "bad" in userland code - not by design

**Comment by zcmgyu** [comment]
@jamuhl Thanks for your reply.

I've already checked it with the non-Gatsby project. The useTranslation hook will not cause any re-render issue. I will try to figure out `gatsby-plugin-react-i18next`. Really thanks. :bow:

https://codesandbox.io/s/why-did-you-render-sandbox-forked-891xf?file=/src/index.js

**Comment by yash14022001** [comment]
I am not sure if the issue is still there or not, but I encountered some bug from my side when I was having the issue of re-rendering.
![image](https://user-images.githubusercontent.com/47099011/231376078-f536db89-5a18-4e1e-9170-2c35f66c18df.png)

I was listening to `lang` in my application through redux and `i18n` was extracted from `useTranslation()` hook. The line `if (i18n.language === lang) return;` was not there when I was having the issue of re-rendering.

But when I added that line, basically avoiding `i18n.setLanguage()` call when language was the same as previous, my re-rendering issue was solved and the `t` hook was working perfectly afterwards.

I am curious why the `setLanguage()` does not apply the same validation check to avoid the duplicate language set, thus avoiding other re-rendering caused by `t` hook change.

Btw I am using `v11.18.3` of the package and using react 17. I am new to react18n's codebase so not know anything about it. So pardon if this is the wrong interpretation of things :)
